### PR TITLE
Contact channel APIs

### PR DESCRIPTION
**Pull request by fomalhautb**

### CODE REVIEW

File: apps/backend/src/app/api/v1/team-memberships/[team_id]/[user_id]/route.tsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import { teamMembershipsCrudHandlers } from "../../crud";
 
+// TODO: move this to /team-memberships
```
Comment (portermichael) [inline apps/backend/src/app/api/v1/team-memberships/[team_id]/[user_id]/route.tsx:3]: What is the repos policy with todos? Do we get to them next time the code with a todo changes? Put an issue together when a todo is created?

Comment (fomalhautb) [reply apps/backend/src/app/api/v1/team-memberships/[team_id]/[user_id]/route.tsx:3]: We have an internal task board to track all these, and we only publish part of the easy-to-understand tasks to the public GitHub issues (this one for example is not super interesting to most of the developers). TODOs in the code are just reminders when we work on the same code.

Comment (N2D4) [reply apps/backend/src/app/api/v1/team-memberships/[team_id]/[user_id]/route.tsx:3]: TODOs are also things that would be useful to someone reading the code later down the line

File: apps/backend/src/app/api/v1/auth/oauth/callback/[provider_id]/route.tsx [hunk 2]
```diff
@@ -268,14 +269,32 @@ const handler = createSmartRouteHandler({
...
+                        projectId: outerInfo.projectId,
+                        type: 'EMAIL',
+                        value: userInfo.email,
+                      }
+                    );
+
+                    if (oldContactChannel && oldContactChannel.usedForAuth) {
+                      primaryEmailAuthEnabled = false;
+                    }
+                    // TODO: check merge accounts
```
Comment (N2D4) [inline apps/backend/src/app/api/v1/auth/oauth/callback/[provider_id]/route.tsx]: what does this mean?

File: apps/e2e/tests/backend/endpoints/api/v1/auth-flows.test.ts
Comment (N2D4) [inline apps/e2e/tests/backend/endpoints/api/v1/auth-flows.test.ts:1]: each of the test files in `endpoints/...` is about a specific endpoint; `auth-flows` is not an endpoint. Seems to me like these tests should be in `sign-in`

File: apps/e2e/tests/backend/endpoints/api/v1/auth-flows.test.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
+import { it } from "../../../../helpers";
+import { Auth, ContactChannels, backendContext, niceBackendFetch } from "../../../backend-helpers";
+
+it("signs in with OTP, disable used for auth, then should not be able to sign in again", async ({ expect }) => {
```
Comment (N2D4) [inline apps/e2e/tests/backend/endpoints/api/v1/auth-flows.test.ts]: test names should be legible with the title of the file as a prefix; "endpoints/api/v1/auth-flows signs in with OTP, disable used for auth, then should not be able to sign in again" is a confusing (and grammatically wrong) sentence

(this also applies to all the other tests)

File: apps/e2e/tests/backend/endpoints/api/v1/auth/otp/sign-in.test.ts [hunk 4]
```diff
@@ -82,13 +84,49 @@ it("should sign in users created with the server API even if sign up is disabled
   `);
 });
 
+it("should not be able to sign in if email is not verified", async ({ expect }) => {
```
Comment (N2D4) [inline apps/e2e/tests/backend/endpoints/api/v1/auth/otp/sign-in.test.ts]: as above; the noun is the endpoint `/endpoints/api/v1/auth/otp/sign-in`

```suggestion
it("should not allow signing in if email is not verified", async ({ expect }) => {
```

File: apps/e2e/tests/backend/endpoints/api/v1/contact-channels/legacy-send-verification-code.test.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
+import { it } from "../../../../../helpers";
+import { Auth, backendContext, niceBackendFetch } from "../../../../backend-helpers";
+
+it("can't send a verification code while logged out", async ({ expect }) => {
```
Comment (N2D4) [inline apps/e2e/tests/backend/endpoints/api/v1/contact-channels/legacy-send-verification-code.test.ts]: as above

```suggestion
it("doesn't send a verification code if logged out", async ({ expect }) => {
```

File: apps/e2e/tests/backend/endpoints/api/v1/auth-flows.test.ts
Comment (fomalhautb) [reply apps/e2e/tests/backend/endpoints/api/v1/auth-flows.test.ts:1]: These test combinations of sign-in flows, basically testing the graph on lucid chart

File: apps/backend/src/app/api/v1/auth/oauth/callback/[provider_id]/route.tsx (hunk 2 above)
Comment (fomalhautb) [reply apps/backend/src/app/api/v1/auth/oauth/callback/[provider_id]/route.tsx]: Just a marker for the future for implementing the account merging feature

Comment (N2D4) [reply apps/backend/src/app/api/v1/auth/oauth/callback/[provider_id]/route.tsx]:
```suggestion
                    // TODO: check whether this OAuth account can be used to login to another existing account instead
```

File: apps/e2e/tests/backend/endpoints/api/v1/auth-flows.test.ts
Comment (N2D4) [reply apps/e2e/tests/backend/endpoints/api/v1/auth-flows.test.ts:1]: we should put them somewhere else then

File: apps/e2e/tests/backend/endpoints/api/v1/auth-flows.test.ts [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import { it } from "../../../../helpers";
 import { Auth, ContactChannels, backendContext, niceBackendFetch } from "../../../backend-helpers";
 
-it("signs in with OTP, disable used for auth, then should not be able to sign in again", async ({ expect }) => {
+it("should not be able to sign in again after signing in with OTP and disabled auth", async ({ expect }) => {
```
Comment (N2D4) [inline apps/e2e/tests/backend/endpoints/api/v1/auth-flows.test.ts]:
```suggestion
it("should not be able to sign in again after signing in with OTP and disabling auth", async ({ expect }) => {
```

File: apps/e2e/tests/backend/endpoints/api/v1/auth-flows.test.ts [hunk 7]
```diff
@@ -65,7 +65,7 @@ it("signs in with password first, then cannot sign in with otp anymore", async (
 });
 
 
-it("signs in with OTP first, then signs in with oauth should give an account with used_for_auth false", async ({ expect }) => {
+it("should sign in with OTP first, then sign in with OAuth and have an account with used_for_auth set to false", async ({ expect }) => {
```
Comment (N2D4) [inline apps/e2e/tests/backend/endpoints/api/v1/auth-flows.test.ts]:
```suggestion
it("signing in with OTP first, then signing in with OAuth, should set used_for_auth to false", async ({ expect }) => {
```

**Pull request merged**